### DISCUSSION TITLE
nginx-proxy with fail2ban - log path

### DISCUSSION THREAD

**Discussion by rucksman** [nginx-proxy/nginx-proxy#1677] [Q&A]
I am using nginx-proxy since quite a long time. The frontend of some of my websites are secured with basic auth. Now I want to block the increasingly annoying attacks on basic auth with fail2ban. The logs from the proxy are written to /dev/stdout and /dev/stderr as usual to access them via docker logs. But for fail2ban I need a (persistent) path to the logs. How can I achive that? Of course I could use something lile /var/lib/docker/containers/*/*-json.log, but that has a lot of drawbacks. Any ideas how I can get persistent logfiles in a persistent volume?

**Comment by Samhamsam** [comment]
Hello, did you find a solution? I have the same question right now.

**Comment by rucksman** [comment]
I think (not sure after 2 months ;-) ) I created an error_log and an access_log in the vhost conf and then made a volume for these log files in my docker-compose file.

**Reply by airdogvan** [reply]
for those of us who are not so familiar with nginx, could you kindly specify where the vhost conf file is located and what you wrote in it?

**Comment by rucksman** [comment]
Inside the container the vhost files reside in /etc/nginx/vhost.d

Just add a conf file there which has the exact same name as the VIRTUAL_HOST (https://github.com/nginx-proxy/nginx-proxy#per-virtual_host).

The entry in this file is quite simple:
`error_log /etc/nginx/log/ot-error.log;`

Then add a volume to your docker-compose file and your are good to go.

**Comment by airdogvan** [comment]
thanks so much, much appreciated!
